app/models/teaching_events/search.rb

```diff
@@ -0,0 +1,67 @@
+module TeachingEvents
+  class Search
+    include ActiveModel::Model
+    include ActiveModel::Validations::Callbacks
+
+    FUTURE_MONTHS = 6
+
+    attr_accessor :postcode, :online, :type, :distance
+
+    validates :postcode, presence: true, postcode: { allow_blank: true, accept_partial_postcode: true }, if: :distance
+
+    before_validation { self.distance = nil if distance.blank? }
+    before_validation { self.postcode = postcode.to_s.strip.upcase.presence }
+
+    def results
+      query.flat_map(&:teaching_events).sort_by(&:start_at)
+    end
+
+  private
+
+    def query(limit: 12)
+      conditions = {
+        type_ids: type_condition,
+        postcode: postcode_condition,
+        online: online_condition,
+        quantity_per_type: limit,
+        radius: distance_condition,
```

Comment:
Following conversation with the Events team, please could we change the radius as follows:

- remove 25 miles
- add 30 miles (this is what they use for their event comms)
- add 50 miles (as a test to see if people want a wider area)